.gitlab-ci.yml

```diff
@@ -14,19 +14,65 @@ variables:
   BUILD_DIR: build
 
 stages:
+  - build
   - test
   - test_acceptance_build
   - test_acceptance
-  - build
   - publish
 
+compile:
+  image: golang:1.14
+  stage: build
+  before_script:
+    - mkdir -p /go/src/$(dirname $REPO_NAME)/mender-cli /go/src/_/builds
+    - cp -r $CI_PROJECT_DIR /go/src/$(dirname $REPO_NAME)
+    - cd /go/src/$(dirname $REPO_NAME)/mender-cli
+    - make get-build-deps
+  script:
+    - make build-multiplatform
+    - cp /go/src/$(dirname $REPO_NAME)/mender-cli/mender-cli.linux.amd64 $CI_PROJECT_DIR/
+    - cp /go/src/$(dirname $REPO_NAME)/mender-cli/mender-cli.darwin.amd64 $CI_PROJECT_DIR/
+  artifacts:
+    paths:
+      - mender-cli.linux.amd64
+      - mender-cli.darwin.amd64
+
+test:smoketests:mac:
+  stage: test
+  dependencies:
+    - compile
+  before_script: []
+  script:
+    - ./mender-cli.darwin.amd64
+  tags:
+    - mender-mac-slave
+
+test:smoketests:linux:
+  image: debian:9
+  stage: test
+  dependencies:
+    - compile
+  before_script: []
+  script:
+    - ./mender-cli.linux.amd64
+
+test:static:  
+  image: golang:1.14
+  stage: test
+  script:
+    - make get-go-tools
+    - make test-static
+
 test_acceptance_build:tools:
   image: golang:1.14
   stage: test_acceptance_build
+  dependencies:
+    - compile
   script:
     - apt-get update && apt-get install make
     # make target env var
     - export PROJECT_DIR="$CI_PROJECT_DIR/"
+    - cp mender-cli.linux.amd64 $CI_PROJECT_DIR/mender-cli
```

Comment:
Can we remove the suffix after moved to the CI_PROJECT_DIR?

Reply:
you mean instead of copying just moving it? the `-cli` suffix in `mender-cli` has to remain for the downstream jobs to work...

Reply:
No, I meant that:
Shouldn't the name of the binary be `mender-cli` after it has been copied? And not `mender-cli.linux.amd64`?

Reply:
yeah... cp takes source first and target second? so... `mender-cli.linux.amd64` is the binary we get from the `compile` job => so call it just `mender-cli`

Reply:
Yeh :)

Reply:
Lol, I misread this. NVM